packages/react-reconciler/src/ReactStrictModeWarnings.js

```diff
@@ -289,6 +296,70 @@ if (__DEV__) {
       });
     }
   };
+
+  ReactStrictModeWarnings.recordLegacyContextWarning = (
+    fiber: Fiber,
+    instance: any,
+  ) => {
+    // Dedup strategy: Warn once per component.
+    if (didWarnAboutLegacyContext.has(fiber.type)) {
+      return;
+    }
```

Comment:
nit: Should the `StrictMode` ancestor check (below) come before this early return?